apps/dashboard/src/components/activity/activity-panel.tsx

```diff
@@ -1,61 +1,74 @@
-import { useEffect, useState } from 'react';
-import { useQueryClient } from '@tanstack/react-query';
+import { IActivityJob } from '@novu/shared';
 import { motion } from 'motion/react';
-import { RiPlayCircleLine, RiRouteFill } from 'react-icons/ri';
-import { IActivityJob, JobStatusEnum } from '@novu/shared';
+import { RiMemoriesFill, RiPlayCircleLine, RiRouteFill } from 'react-icons/ri';
+import { Tooltip, TooltipContent, TooltipTrigger } from '../primitives/tooltip';
 
-import { ActivityJobItem } from './activity-job-item';
-import { InlineToast } from '../primitives/inline-toast';
-import { useFetchActivity } from '@/hooks/use-fetch-activity';
-import { ActivityOverview } from './components/activity-overview';
+import { useEnvironment } from '@/context/environment/hooks';
+import { useTriggerWorkflow } from '@/hooks/use-trigger-workflow';
+import { useTelemetry } from '../../hooks/use-telemetry';
+import { TelemetryEvent } from '../../utils/telemetry';
 import { cn } from '../../utils/ui';
+import { CompactButton } from '../primitives/button-compact';
+import { InlineToast } from '../primitives/inline-toast';
 import { Skeleton } from '../primitives/skeleton';
-import { QueryKeys } from '@/utils/query-keys';
-import { useEnvironment } from '@/context/environment/hooks';
+import { showErrorToast, showSuccessToast } from '../primitives/sonner-helpers';
+import { ActivityJobItem } from './activity-job-item';
+import { ActivityOverview } from './components/activity-overview';
+import { useActivityByTransaction, useActivityPolling } from './hooks';
 
 export interface ActivityPanelProps {
   activityId?: string;
+  transactionId?: string;
   onActivitySelect: (activityId: string) => void;
   headerClassName?: string;
   overviewHeaderClassName?: string;
 }
 
 export function ActivityPanel({
   activityId,
+  transactionId: initialTransactionId,
   onActivitySelect,
   headerClassName,
   overviewHeaderClassName,
 }: ActivityPanelProps) {
-  const queryClient = useQueryClient();
-  const [shouldRefetch, setShouldRefetch] = useState(true);
+  const track = useTelemetry();
+  const { isLoadingTransaction, setTransactionId } = useActivityByTransaction({
+    transactionId: initialTransactionId,
+    onActivityFound: onActivitySelect,
+  });
+
+  const { activity, isPending, error } = useActivityPolling({
+    activityId,
+  });
+
```

Comment:
Moved the logic of fetching and refetching by transaction id to the activity panel to allow initializing by either activityId or transaction id

Reply:
Can you please elaborate on why do we need to do both? The transaction is identified by transactionID, so I think we can just use that.

Reply:
When vieweing an item from the activity feed, we want to view a specific activityId, transacitonId is used in cases of a re-trigger, and also from the test workflow sidebar. As a trigger can potentially generate multiple activity ids, when testing we assume it generates one and just take the first.

Reply:
I see. Let's continue [here](https://github.com/novuhq/novu/pull/7459#discussion_r1923501849). My suggestion is to work with the notification list (activity feed) page as the main resource and the selected notification (details of a run).

If not mistaken, the transactionId for the rerun should be available in the selected notification resource.